lib/ansible/modules/network/opx/opx_cps.py

```diff
@@ -0,0 +1,365 @@
+     networking device running OpenSwitch (OPX). It uses the YANG models
+     provided in https://github.com/open-switch/opx-base-model.
+options:
+  module_name:
+    description:
+      - Yang path to be configured.
+  attr_type:
+    description:
+      - Attribute Yang type.
+  attr_data:
+    description:
+      - Attribute Yang path and thier correspoding data.
+  operation:
+    description:
+      - Operation to be performed on the object.
+    default: create
+    choices: ['delete', 'create', 'set', 'action', 'get']
+  db:
+    description:
+      - Queries/Writes the specified yang path from/to the db.
+    type: bool
+    default: 'no'
+  qualifier:
+    description:
+      - A qualifier provides the type of object data to retrieve or act on.
+    default: target
+    choices: ['target', 'observed', 'proposed', 'realtime', 'registration', 'running', 'startup']
+  commit_event:
+    description:
+      - Attempts to force the auto-commit event to the specified yang object.
+    type: bool
+    default: 'no'
+"""
+
+EXAMPLES = """
+- name: Create VLAN
+  opx_cps:
+    module_name: "dell-base-if-cmn/if/interfaces/interface"
+    attr_data: {
+         "base-if-vlan/if/interfaces/interface/id": 230,
+         "if/interfaces/interface/name": "br230",
+         "if/interfaces/interface/type": "ianaift:l2vlan"
+    }
+    operation: "create"
+- name: Get VLAN
+  opx_cps:
+    module_name: "dell-base-if-cmn/if/interfaces/interface"
+    attr_data: {
+         "if/interfaces/interface/name": "br230",
+    }
+    operation: "get"
+"""
+
+RETURN = """
+response:
+  description: Output from the CPS transaction.
+  returned: always
+  type: dict
+  sample: {'...':'...'}
+changed:
+  description: Returns if the CPS transaction was performed.
+  returned: when a CPS traction is performed.
+  type: dict
+  sample: {'...':'...'}
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.six import iteritems
+
+try:
+    import cps
+    import cps_object
+    import cps_utils
+    HAS_CPS = True
+except ImportError:
+    HAS_CPS = False
+
+
+def get_config(module):
+
+    qualifier = module.params['qualifier']
+    module_name = module.params['module_name']
+    attr_type = module.params["attr_type"]
+    attr_data = module.params["attr_data"]
+    db = module.params["db"]
+    commit_event = module.params["commit_event"]
+    config = dict()
+
+    configobj = parse_cps_parameters(module_name, qualifier,
+                                     attr_type, attr_data, "get",
+                                     db, commit_event)
+    cpsconfig = cps_get(configobj)
+
+    if cpsconfig['response']:
+        for key, val in iteritems(cpsconfig['response'][0]['data']):
+            if key == 'cps/key_data':
+                config.update(val)
+            else:
+                config[key] = val
+
+    return config
+
+
+def diff_dict(base_data, compare_with_data):
```

Comment:
Can the `dict_diff` utils function https://github.com/ansible/ansible/blob/devel/lib/ansible/module_utils/network/common/utils.py#L214 reused instead of writing a new one?